lib/meta-attachments.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Class for adding an attachment meta box with custom cropping.
+ *
+ * @package Course Maker Pro
+ * @author  thebrandiD
+ * @license GPL-2.0+
+ * @link    https://thebrandidthemes.com/
+ */
+
+ /**
+  * Class for handling meta attachments.
+  */
+class Meta_Attachments {
+
+	/**
+	 * Unique ID for this meta box.
+	 *
+	 * @var string $id
+	 */
+	private string $id = '';
+
+	/**
+	 * Unique slug for the meta box (see sanitize_title).
+	 *
+	 * Can be the same as $id.
+	 *
+	 * @var string $slug
+	 */
+	private string $slug = '';
+
+	/**
+	 * Meta Key to store the data in post meta (use an underline to make the custom field private).
+	 *
+	 * @var string $meta_key
+	 */
+	private string $meta_key = '';
+
+	/**
+	 * Array of supported post types.
+	 *
+	 * @var array
+	 */
+	private array $post_types = array();
+
+	/**
+	 * The label that will display inside the meta box.
+	 *
+	 * @var string $meta_label
+	 */
+	private string $meta_label = '';
+
+	/**
+	 * The priority of where to show the meta box (see add_meta_box).
+	 *
+	 * @var string $meta_priority
+	 */
+	private string $meta_priority = 'high';
+
+	/**
+	 * Meta location within the editor (see add_meta_box).
+	 *
+	 * @var string $meta_location
+	 */
+	private string $meta_location = 'side';
+
+	/**
+	 * The meta add attachment button label.
+	 *
+	 * @var string $attachment_button_label
+	 */
+	private string $attachment_button_label = '';
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param array $args {
+	 *     An array of arguments.
+	 *
+	 *     @type string $meta_key Meta key to register (stored in post meta)
+	 *     @type array $post_types Array of post types to list this meta box on.
+	 *     @type string $meta_label Label for insert attachment button (see add_meta_box).
+	 *     @type string $meta_priority Priority of meta box (see add_meta_box).
+	 *     @type string $meta_location Location of meta box (see add_meta_box).
+	 * }
+	 */
+	public function __construct( $args ) {
+		// Set defaults.
+		$defaults = array(
+			'id' => 'course_maker_featured_article_image',
+			'meta_key' => 'course_maker_featured_article_image',
+			'post_types' => array(
+				'post',
+			),
+			'meta_label' => __( 'Add Image', 'coursemaker' ),
+			'meta_priority' => 'high',
+			'meta_location' => 'side',
+			'attachment_button_label' => __( 'Add Image', 'coursemaker' ),
+		);
+
+		// Merge arguments.
+		$args = wp_parse_args( $args, $defaults );
+
+		// Set class variables.
+		foreach ( $args as $key => $value ) {
+			$this->{ $key } = $value;
+		}
+	}
+
+	/**
+	 * Run the hooks associated with this class.
+	 */
+	public function run_hooks() {
```

Comment:
This adds the actions for the metaboxes with callback. 

Finally we load the media scripts needed.